backends/mock_client.go

```diff
@@ -0,0 +1,52 @@
+package backends
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+type MockReturnErrorBackend struct{}
+
+func (b *MockReturnErrorBackend) Get(ctx context.Context, key string) (string, error) {
+	return "", fmt.Errorf("This is a mock backend that returns this error on Get() operation")
+}
+
+func (b *MockReturnErrorBackend) Put(ctx context.Context, key string, value string, ttlSeconds int) error {
+	return fmt.Errorf("This is a mock backend that returns this error on Put() operation")
+}
+
+func NewErrorReturningBackend() *MockReturnErrorBackend {
+	return &MockReturnErrorBackend{}
+}
+
+type MockDeadlineExceededBackend struct{}
```

Comment:
This doesn't use a mock framework, so please consider renaming it.

Reply:
Good catch. Changed to `deadlineExceedingBackend`